rcl/src/rcl/publisher.c

```diff
@@ -421,7 +421,11 @@ rcl_publisher_is_valid(const rcl_publisher_t * publisher)
     return false;  // error already set
   }
   if (!rcl_context_is_valid(publisher->impl->context)) {
-    RCL_SET_ERROR_MSG("publisher's context is invalid");
+    if (!rcl_error_is_set()) {
+      // rcl_context_is_valid can return false both in the error case, and when the context
+      // hasn't been initialized.  It will only set the error message in the first case.
+      RCL_SET_ERROR_MSG("publisher's context is invalid");
+    }
```

Comment:
i would do the following only to use the designed interfaces. (relying on the implementation behavior could be changed, for example we could add the error message if the context id is zero in the future? that breaks this if statement.)

```suggestion
    RCL_SET_ERROR_MSG("publisher's context is either invalid or uninitialized");
```

Reply:
thanks for the explanation.